luaJIT uses __register_frame and  __deregister_frame
our clang only links libgcc_s for c++ by default

diff -wpruN --no-dereference '--exclude=*.orig' a~/cmake.deps/cmake/BuildLuajit.cmake a/cmake.deps/cmake/BuildLuajit.cmake
--- a~/cmake.deps/cmake/BuildLuajit.cmake	1970-01-01 00:00:00
+++ a/cmake.deps/cmake/BuildLuajit.cmake	1970-01-01 00:00:00
@@ -24,6 +24,8 @@ else()
 endif()
 if(CMAKE_SYSTEM_NAME MATCHES "OpenBSD")
   set(AMD64_ABI "LDFLAGS=-lpthread -lc++abi")
+elseif(CMAKE_SYSTEM_NAME MATCHES "SunOS")
+  set(AMD64_ABI "LDFLAGS=-lgcc_s")
 else()
   set(AMD64_ABI "")
 endif()
